### PR TITLE
Add support for JSON output (--json)

### PR DESCRIPTION
**Pull request by zmwangx**
<s>Blocked on #74 (this branch is based on #74).</s>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -157,6 +157,8 @@ Please substitute `$version` with the appropriate package version.
       -t dN, --time dN      time limit search [h5 (5 hrs), d5 (5 days), w5 (5
                             weeks), m5 (5 months), y5 (5 years)]
       -w SITE, --site SITE  search a site using Google
+      --json                output in JSON format; implies --noprompt
+      --noninteractive      perform search and exit, do not prompt for further
```
Comment (jarun) [inline README.md]: Typo perhaps? As per our earlier discussion we are not going to have only `--noprompt`.

File: googler [hunk 2]
```diff
@@ -646,6 +658,9 @@ debug = args.debug
 keywords = args.keywords
 insite = args.insite
 noninteractive = args.noninteractive
+if args.json:
+    output_format = 'json'
+    noninteractive = True
```
Comment (jarun) [inline googler]: should we couple json and noninteractive? Users can use both the options together if they ust want to dump the results of a single fetch.

Comment (zmwangx) [inline googler]: JSON output is most likely piped, I don't think anyone would rather read JSON in interactive use. Also, our prompt uses `raw_input` (python2) or `input` (python3) which writes stdout, totally screwing up piping.

There's always `--count` and `--start` to handle pagination.

File: README.md (hunk 1 above)
Comment (zmwangx) [inline README.md]: I wrote this feature before changing the option name. This is an oversight during rebasing.

Comment (zmwangx) [inline README.md]: Fixed.

File: googler (hunk 2 above)
Comment (jarun) [inline googler]: :+1:

**Comment by zmwangx** [comment]
Ping. Any other comments?

**Comment by jarun** [comment]
Nope. Sorry I almost forgot about this PR.

**Comment by jarun** [comment]
Merged. Thanks!

**Comment by zmwangx** [comment]
No problem. I'll add it to the completions.

**Pull request merged**